stablehlo/tests/CheckOps.cpp

```diff
@@ -156,6 +157,36 @@ llvm::Error evalExpectSerializedEqOp(const Tensor &expected, StringRef probeId,
   return evalExpectEqOp(expected, *tensor);
 }
 
+llvm::Error evalExpectIsCloseOp(const Tensor &lhs, const Tensor &rhs,
+                                const Tensor &abs_error, const Tensor &input) {
+  if (!isSupportedFloatType(abs_error.getElementType())) {
+    const std::string abs_errorType = debugString(abs_error.getType());
+    return invalidArgument("abs_error must be a float tensor, got %s",
+                           abs_errorType.c_str());
+  }
+  std::string mismatches;
+  llvm::raw_string_ostream output(mismatches);
+  for (auto lhsIt = lhs.index_begin(), rhsIt = rhs.index_begin(),
+            abs_errorIt = abs_error.index_begin(),
+            inputIt = input.index_begin();
+       lhsIt != lhs.index_end(); ++lhsIt, ++rhsIt, ++abs_errorIt, ++inputIt) {
+    double abs_error_ =
+        abs_error.get(*abs_errorIt).getFloatValue().convertToDouble();
+    if (!areApproximatelyEqual(lhs.get(*lhsIt), rhs.get(*rhsIt), abs_error_)
+             .getBooleanValue()) {
+      output << "\n  index=" << (*lhsIt) << ", actual=" << lhs.get(*lhsIt)
+             << ", expected=" << rhs.get(*rhsIt)
+             << ", atol=" << abs_error.get(*abs_errorIt)
+             << ", input=" << input.get(*inputIt);
```

Comment:
It is not clear from the error message how the `input` value to be interpreted. Imagine we get this error, the user will have no idea what function  (asin in this case) this input is executed with to generate the actual value.  Without that information,  the value of providing input here is still unclear to me.

Reply:
I assumed that functions under stablehlo/tests/ are for testing purposes and not exposed to the end-users. When testing, one always knows what functions are being tests.

If this assumption is incorrect, what would be the right place to implement this functionality of `check_is_close` that is used only for testing math functions?

Btw, when users see the above error message, the interpretation of values of `lhs`, `rhs`, and `input` would be all equally unclear/useless when the function being tested is unknown.

Reply:
Since `check.expect_is_close_to_reference` is used only in scripts that test specific math functions, there is no ambiguity of interpreting the `input` value in the error messages.